crates/core/tedge_agent/src/agent.rs

```diff
@@ -171,6 +171,9 @@ impl SmAgent {
     pub async fn start(&mut self) -> Result<(), AgentError> {
         info!("Starting tedge agent");
 
+        let mqtt = Client::connect(self.name.as_str(), &self.config.mqtt_client_config).await?;
+        let mut operations = mqtt.subscribe(self.config.request_topics.clone()).await?;
```

Comment:
Not quite sure if this fix is sufficient (although I agree that this might make the issue less frequent). What if the mapper process sends the request before this agent process execution even reaches this line? That message will still be lost, right?(unless we're sending those requests with retained flag ON, which I doubt).

Reply:
Agree that there are still two race conditions here but these are not related to the retain flag.

The first race condition arises if this is the very first connection of the agent and if messages have already been sent to the input topic of the agent. Since no client has subscribed to this topic the broker discards all these messages. To fix that subscriptions must created on behalf of the agent before any message is sent. This can be done on install for all the daemon.

The second issue is what is mitigated by this PR. With the current `mqtt_client` crate, a client connection might start to consume messages before a call to subscribe. The proper fix is a bit more complex (see #575).

Reply:
> To fix that subscriptions must be created on behalf of the agent before any message is sent. This can be done on install for all the daemon.

Understood.